mlflow/pyfunc/__init__.py

```diff
@@ -395,7 +395,15 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
 
         :param data: Model input as one of pandas.DataFrame, numpy.ndarray,
                      scipy.sparse.(csc.csc_matrix | csr.csr_matrix), List[Any], or
-                     Dict[str, numpy.ndarray]
+                     Dict[str, numpy.ndarray].
+                     For model signatures with tensor spec inputs
+                     (e.g. the Tensorflow core / Keras model), the input data type must be one of
+                     `numpy.ndarray`, `List[numpy.ndarray]`, `Dict[str, numpy.ndarray]` or
+                     `pandas.DataFrame`. If data is of `pandas.DataFrame` type and an input field
+                     requires multidimensional array input, the corresponding column values in
```

Comment:
I would prefer to not change the pyfunc predict function. the correct way to call pyfunc with multidimensional tensors is to pass in numpy array or a dict of numpy arrays. This is not possible in spark, so we should fix it in there. Can we only make the reshaping work in spark_udf code path?

Reply:
My purpose is (1) in https://github.com/mlflow/mlflow/pull/7291#issuecomment-1324434216
but, final decision is up to you.